docs/lib/manual/Flash.md

```diff
@@ -0,0 +1,142 @@
+
+```
+***********************************************************
+                    BUS FAULT
+***********************************************************
+
+* Active Box ID: 1
+* FAULT SYNDROME REGISTERS
+
+  CFSR: 0x00008200
+  BFAR: 0x40020007
+  --> PRECISERR: precise data access.
+
+* No MPU violation found
+
+* EXCEPTION STACK FRAME
+
+  lr: 0xFFFFFFFD
+  --> FP stack frame not saved.
+  --> Exception from NP mode.
+  --> Return to PSP.
+
+  sp: 0x1FFF34D0
+      sp[07]: 0x21000000 | xPSR
+      sp[06]: 0x0000D6FE | pc
+      sp[05]: 0x0000D6F1 | lr
+      sp[04]: 0x00000000 | r12
+      sp[03]: 0x40020000 | r3
+      sp[02]: 0x00000009 | r2
+      sp[01]: 0x00000000 | r1
+      sp[00]: 0x00000000 | r0
+
+HALT_ERROR(./core/vmpu/src/mpu_kinetis/vmpu_kinetis.c#166): Access to restricted resource denied.
+```
+
+## Configuring the ACL for the FTFE driver
+
+To find out which peripheral is responsible for this fault, you can look in the [peripheral access layer](https://github.com/ARMmbed/mbed-os/blob/e5ba1d2/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/device/MK64F12.h) for your development board. If you search here for `40020000` - peripherals are normally aligned - you find the `FTFE_BASE` address, which is the flash driver for the K64F.
+
+```cpp
+/** Peripheral FTFE base address */
+#define FTFE_BASE                                (0x40020000u)
+```
+
+You can enable this peripheral for the secure box by adding the following definition to the ACL list:
+
+```cpp
+    {FTFE,             sizeof(*FTFE),  UVISOR_TACLDEF_PERIPH},
+```
+
+After re-compiling the code, you can now see - over the serial port - that you can successfully write to flash. However, uVisor still halts when trying to read from flash.
+
+```
+***********************************************************
+                    BUS FAULT
+***********************************************************
+
+* Active Box ID: 1
+* FAULT SYNDROME REGISTERS
+
+  CFSR: 0x00008200
+  BFAR: 0x000FF000
+  --> PRECISERR: precise data access.
+
+* MPU FAULT:
+  Slave port:       0
+  Address:          0x000FF000
+  Faulting regions:
+    R00: 0x00000000 0xFFFFFFFF 0x000007D0 0x00000001
+  Master port:      0
+  Error attribute:  Data READ (user mode)
+
+HALT_ERROR(./core/vmpu/src/kinetis/vmpu_kinetis.c#144): Access to restricted resource denied
+```
+
+## Configuring the ACL to read from flash
+
+As you can see from the log message, the MPU prevents reading from address `0x000FF000`. This is the value of the `address` you try to read. The K64F uses a memory-mapped address space to access flash storage which is mapped from `0x0` to `0x100000`, and you're using a sector size of `0x1000` bytes. To get around this you can add this specific part of the address space to the ACL list. A nice addition is that you can limit boxes to only part of the flash storage. It's not all or nothing.
```

Comment:
Please add a comma after "to access flash storage". Please add a comma after "To get around this".